### DISCUSSION TITLE
Can two peers connect through a third peer?

### DISCUSSION THREAD

**Discussion by LinuxinaBit** [markqvist/Reticulum#422] [Help & Questions] [answered]
This is a simple-ish question, but I’ve confused myself by thinking about LXMF and Reticulum like they’re a combined project.
I just want to make sure I haven’t been misunderstanding a fundamental part of Reticulum.

I know propagation nodes are a LXMF specific thing, but how does Reticulum handle routing on its own?
Does it require a transport node to carry communications between two otherwise unconnected peers, or can those two peers communicate directly through a third peer?

I’ve read the docs and they say, “Reticulum can function both with and without infrastructure. When transport nodes are available, they can route traffic over multiple hops for other nodes, and will function as a distributed cryptographic keystore. When there is no transport nodes available, all nodes that are within communication range can still communicate.”
I think that means that two peers can’t just randomly decide to communicate through a third peer, for example:

These work:
Peer -> Transport Node -> Peer
Peer -> Peer
This does not:
Peer -> Peer -> Peer

If this is correct, why is Reticulum implemented like this? My intuition says that all peers should route traffic, but the docs say, “Every node can become a transport node, simply by enabling it in it’s configuration, but there is no need for every node on the network to be a transport node. Letting every node be a transport node will in most cases degrade the performance and reliability of the network.”
Why is that?
Thank you in advance for clearing this up :)

**Comment by markqvist** [comment, marked as answer]
Ok, here's a *very* short and simplified version.

Routing traffic is not free, it requires some sort of knowledge, and obtaining that knowledge has a bandwidth cost associated with it, no matter how smart you get about it. At the bottom of it, information theory is really thermodynamics, and you don't get a free meal in terms of energy. Information is energy, so by obtaining it, there will *always* be expenditure. In any network, that expenditure is bandwidth (which is, again, energy).

1. Let all nodes route traffic, and you end up wasting huge amounts of bandwidth on path/routing/maintenance traffic for potentially hundreds of nodes that will never actually route any traffic. And in addition, routing efficiency will be much worse, most packets will likely need many more hops than optimal.
2. You design some algorithm that tries to elect optimal routing topologies. You are now wasting all your bandwidth on that. There's a physical limit too, in terms of how many nodes your network can now support, which gets exponentially smaller when your medium speed drops. Goodbye to using LoRa and similar.
3. You then say, to hell with it, and just decide that everyone flood fills everything, and hope for the best. Well, you see what happens here too.

In the end, the approach Reticulum takes is the most optimal, and the only solution that is scalable and functional in the real world, even over very low-bandwidth mediums.
